src/RestApi/StoreApi/Schemas/ProductSchema.php

```diff
@@ -254,27 +260,41 @@ protected function get_properties() {
 	 */
 	public function get_item_response( $product ) {
 		return [
-			'id'             => $product->get_id(),
-			'name'           => $product->get_title(),
-			'variation'      => $product->is_type( 'variation' ) ? wc_get_formatted_variation( $product, true, true, false ) : '',
-			'permalink'      => $product->get_permalink(),
-			'sku'            => $product->get_sku(),
-			'description'    => apply_filters( 'woocommerce_short_description', $product->get_short_description() ? $product->get_short_description() : wc_trim_string( $product->get_description(), 400 ) ),
-			'on_sale'        => $product->is_on_sale(),
-			'prices'         => $this->get_prices( $product ),
-			'average_rating' => $product->get_average_rating(),
-			'review_count'   => $product->get_review_count(),
-			'images'         => ( new ProductImages() )->images_to_array( $product ),
-			'has_options'    => $product->has_options(),
-			'is_purchasable' => $product->is_purchasable(),
-			'is_in_stock'    => $product->is_in_stock(),
-			'add_to_cart'    => [
+			'id'                  => $product->get_id(),
+			'name'                => $product->get_title(),
+			'variation'           => $product->is_type( 'variation' ) ? wc_get_formatted_variation( $product, true, true, false ) : '',
+			'permalink'           => $product->get_permalink(),
+			'sku'                 => $product->get_sku(),
+			'description'         => apply_filters( 'woocommerce_short_description', $product->get_short_description() ? $product->get_short_description() : wc_trim_string( $product->get_description(), 400 ) ),
+			'on_sale'             => $product->is_on_sale(),
+			'prices'              => $this->get_prices( $product ),
+			'average_rating'      => $product->get_average_rating(),
+			'review_count'        => $product->get_review_count(),
+			'images'              => ( new ProductImages() )->images_to_array( $product ),
+			'has_options'         => $product->has_options(),
+			'is_purchasable'      => $product->is_purchasable(),
+			'is_in_stock'         => $product->is_in_stock(),
+			'low_stock_remaining' => $this->get_low_stock_remaining( $product ),
+			'add_to_cart'         => [
 				'text'        => $product->add_to_cart_text(),
 				'description' => $product->add_to_cart_description(),
 			],
 		];
 	}
 
+	/**
+	 * If a product has low stock, return the remaining stock amount for display.
+	 *
+	 * @param \WC_Product|\WC_Product_Variation $product Product instance.
+	 * @return integer|null
+	 */
+	protected function get_low_stock_remaining( $product ) {
```

Comment:
Should we start typehinting arguments?  Example:

```php
protected function get_low_stock_remaining( \WC_Product $product )
```

If the argument is we don't want to fatal if it's not instance of `WC_Product` (or child instance) then is that also argument for ensuring `$product` is of the correct type before using it in the function logic? 

I'm not going to be dogmatic about this (yet), but I do think now that we're using more modern PHP, that we should also start adopting best practices for type-hinting etc as well.

Reply:
I read that PHP5 only has partial support for typehinting, and you cannot hint scalar types. We can start using it as long as our unit tests will catch incompatibilities :)